sycl/source/detail/queue_impl.hpp

```diff
@@ -404,7 +404,14 @@ class queue_impl {
     handler Handler(Self, MHostQueue);
     Handler.saveCodeLoc(Loc);
     CGF(Handler);
+    // Scheduler will later omit events, that are not required to execute tasks.
+    // Host and interop tasks, however, are not submitted to low-level runtimes
+    // and require separate dependency management.
+    if (has_property<property::queue::in_order>())
```

Comment:
Is it necessary to do this for all types of commands?
How does it play with https://github.com/intel/llvm/blob/sycl/sycl/source/detail/scheduler/commands.cpp#L508 ?

Reply:
I think it is. Command group is not ready unless we call finalize, and after that there's no way we can add a new dependency.

> How does it play with https://github.com/intel/llvm/blob/sycl/sycl/source/detail/scheduler/commands.cpp#L508 ?

Thanks, I forgot to commit those changes.

Reply:
Can't we check if handler::MType if a host task right here?